### ISSUE TITLE
loading indicator

### ISSUE THREAD

**Issue by johntseng** [ThirdCornerLLC/tc-grid#19]
Can we show something to indicate that the grid is loading?

**Comment by Zacharias3690** [comment]
I think we could do this with some sort of callback passed in to onPage/SortChange similar to what gulp does to end tasks.

**Comment by timgit** [comment]
Agreed. We could just add an optional callback to these methods and a config setting to turn on a loading animation.  The visualization could be a configurable class that's added/removed so it can be customized.  So now the only question is which animated gif are we going to ship for this? :)

**Comment by Zach-Barnes-Volusion** [comment]
Where's a good position to place the visual? Anywhere on the table will look awkward, and outside of the table is outside the scope of tc-grid

**Comment by goofygrin** [comment]
the whole grid going to a spinner?

**Comment by Zacharias3690** [comment]
What about graying out the grid and overlaying a loading indicator in the middle of it?

**Comment by goofygrin** [comment]
That's what I was suggesting.

**Comment by Zacharias3690** [comment]
Oh okay, I thought you mean replacing the grid with a loading indicator.

**Comment by goofygrin** [comment]
That's an implementation detail ;)

**Comment by timgit** [comment]
My 2 cents opinion on this would be a pure CSS approach if possible.  If the hosting app has a standard for their loading indicators, and if it were CSS as well, they could just give us their class to override our default class.

**Comment by goofygrin** [comment]
agreed.

**Comment by Zacharias3690** [comment]
Agreed. However, I'm not sure how we can do a pure unicode loading
indicator.

**Comment by goofygrin** [comment]
You can have it be a background image...

**Comment by Zacharias3690** [comment]
Yes, but that's not a pure CSS implementation ;)

**Comment by goofygrin** [comment]
Sure it is... it's just a class that someone puts together...  We're not
expecting an image.

**Comment by timgit** [comment]
data URIs ftw
